Puzzlepart/did
# .nvmrc (12) for automatic node-versioning

### Your checklist for this pull request
 
### Review checklist

### Review theme song
🎵  [nervous_testpilot - Wishbone](https://open.spotify.com/track/2yKHHjOqWDrfRb0pHkdSLv) 🎵 

### Description
.nvmrc file which specifies node version 12 (12.18 or 12.19 both work)
### Why?
Because I have several projects using other node versions, e.g. 10, so running `npm run script watch` usually always bombs for me since i forget to run `nvm use 12`

### How to test
1. Check out locally with [gh](https://github.com/cli/cli)
2. Make sure [Node Version Manager](https://github.com/nvm-sh/nvm) is installed on your machine and available as `nvm` in your shell
3. put the following lines in your .zshrc after nvm initialization (i.e. after `"$NVM_DIR/nvm.sh"`)
```
autoload -U add-zsh-hook
load-nvmrc() {
  if [[ -f .nvmrc && -r .nvmrc ]]; then
    nvm use
  elif [[ $(nvm version) != $(nvm version default)  ]]; then
    echo "Reverting to nvm default version"
    nvm use default
  fi
}
add-zsh-hook chpwd load-nvmrc
load-nvmrc
```
4. reload zsh (`source ~/.zshrc`)
5. cd to a different directory than /did
6. cd back into /did
7. **observe that node version is set to 12 automatically**
8. cd back out of /did
9. **observe that nvm now reverts to whatever you have as default node version (e.g. 10)**
🕺
